langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/VertexAiGeminiStreamingChatModelIT.java

```diff
@@ -367,4 +387,123 @@ void should_accept_function_call() {
         System.out.println("Answer: " + weatherResponse.content().text());
         assertThat(weatherResponse.content().text()).containsIgnoringCase("sunny");
     }
+
+    static class StockInventory {
+        @Tool("Get the stock inventory for a product identified by its ID")
+        public int getStockInventory(@P("ID of the product") String product) {
+            if (product.equals("ABC")) {
+                return 10;
+            } else if (product.equals("XYZ")) {
+                return 42;
+            } else {
+                return 0;
+            }
+        }
+    }
+
+    interface StreamingStockAssistant {
+        @dev.langchain4j.service.SystemMessage(
+            "You MUST call `getStockInventory()` for stock inventory requests.")
+        TokenStream chat(String msg);
+    }
+
+    @Test
+    void should_work_with_parallel_function_calls() throws InterruptedException, ExecutionException, TimeoutException {
+        // given
+        VertexAiGeminiStreamingChatModel model = VertexAiGeminiStreamingChatModel.builder()
+            .project(System.getenv("GCP_PROJECT_ID"))
+            .location(System.getenv("GCP_LOCATION"))
+            .modelName("gemini-1.5-flash-001")
+            .logRequests(true)
+            .logResponses(true)
+            .temperature(0.0f)
+            .topK(1)
+            .build();
+
+        StreamingStockAssistant assistant = AiServices.builder(StreamingStockAssistant.class)
+            .streamingChatLanguageModel(model)
+            .chatMemory(MessageWindowChatMemory.withMaxMessages(10))
+            .tools(new StockInventory())
+            .build();
+
+        // when
+        CompletableFuture<Response<AiMessage>> future = new CompletableFuture<>();
+        assistant.chat("Is there more stock of ABC or of XYZ?")
+            .onNext(System.out::println)
+            .onComplete(future::complete)
+            .onError(future::completeExceptionally)
+            .start();
+        Response<AiMessage> response = future.get(30, TimeUnit.SECONDS);
+
+        // then
+        assertThat(response.content().toString()).contains("XYZ");
```

Comment:
This does not seem to verify that functions were called in parallel, I would suggest checking internals of a `ChatMemory`

Reply:
I'm adding a bunch of asserts on the `ChatMemory`.
However, even for the very same scenario, I don't always get parallel tool execution requests 100% of the time. Sometimes the model ask for them serially instead. So this test could be a bit flaky at times, when asserting that there really are parallel calls happening. That's something to be aware of.
But also, what's cool is that it shows that LangChain4j makes this transparent: whether requests are sequential or parallel, the end result is the same!

Reply:
Thanks! If the test is working 90% of the time, it is good enough, we can think of adding automatic retries for flaky tests.
But most important IMHO is that desired behavior is tested.